### ISSUE TITLE
Mixup 구현 시 주의할 점 : input과 output의 dimension

### ISSUE THREAD

**Issue by zooyeonii** [boostcampaitech3/level1-image-classification-level1-cv-06#20]
Mixup을 구현할 때 input의 label은 one-hot vector여야 한다. 
dataset의 label이 class number이거나, model의 output이 softmax vector일 경우 dimension의 수정 또는 클래스 결과값 변환이 필요하다. 

```
for idx, train_batch in enumerate(train_loader):
            inputs, label = train_batch
            
            labels = torch.stack([torch.nn.functional.one_hot(i, num_classes=18) for i in label], dim=0)
            inputs = inputs.to(device)
            labels = labels.to(device)
            
            inputs, labels_a, labels_b, lam = mixup_data(inputs, labels, 0.2, use_cuda)
            inputs, labels_a, labels_b = map(Variable, (inputs, labels_a, labels_b))
            
            outs = model(inputs)
         
            loss = mixup_criterion(criterion, outs, labels_a, labels_b, lam)
            loss_value += loss.item()
            _, preds = torch.max(outs.data, 1)
            
            matches += (lam*preds.eq(torch.argmax(labels_a, dim=1)).cpu().sum().float() 
                        + (1-lam)*preds.eq(torch.argmax(labels_b, dim=1)).cpu().sum().float())
            
            
            optimizer.zero_grad()
            loss.backward()
            optimizer.step()
```
본래 train data의 label은 0-17이므로, `labels = torch.stack([torch.nn.functional.one_hot(i, num_classes=18) for i in label], dim=0) ` 를 통해 one-hot vector로 전환하는 과정이 필요하다. 
prediction (preds)은 softmax vector이고, 클래스를 반환하기 위해 `torch.max(outs.data, 1)`을 통해 가장 큰 값의 index를 반환한다. 
이제 preds는 0-17 class number를 갖게 될 것이고,  `preds.eq(torch.argmax(labels_a, dim=1))` 는 예측값과 ground truth가 같은지 확인하는 코드이다. 
여기서 `torch.argmax(label, dim=1)` 을 살펴보면, 현재 데이터가 batch로 들어오므로, shape은 (batch_size, label len)이다. 따라서 dim=1의 위치에서 argmax를 통해 클래스를 추출해야 한다.
